rts/Lua/LuaHandle.cpp

```diff
@@ -1443,6 +1443,60 @@ void CLuaHandle::UnitHarvestStorageFull(const CUnit* unit)
 	UnitCallIn(cmdStr, unit);
 }
 
+/*** Called when a unit starts it's self destruct command.
+ *
+ * @function Callins:UnitSelfDestructStarted
+ * @param unitID integer
+ * @param unitDefID integer
+ * @param unitTeam integer
+ */
+void CLuaHandle::UnitSelfDestructStarted(const CUnit* unit) 
+{
+	static const LuaHashString cmdStr(__func__);
+	UnitCallIn(cmdStr, unit);
+}
+
+/*** Called when a unit cancel's it's self destruct command.
+ *
+ * @function Callins:UnitSelfDestructCancelled
+ * @param unitID integer
+ * @param unitDefID integer
+ * @param unitTeam integer
+ */
+void CLuaHandle::UnitSelfDestructCancelled(const CUnit* unit) 
+{
+	static const LuaHashString cmdStr(__func__);
+	UnitCallIn(cmdStr, unit);
+}
+
+/*** Called every second while a unit has a self destruct command active.
+ *
+ * @function Callins:UnitSelfDestructProgress
+ * @param unitID integer
+ * @param unitDefID integer
+ * @param unitTeam integer
+ * @param remainingSeconds float
+ */
+void CLuaHandle::UnitSelfDestructProgress(const CUnit* unit, float remainingSeconds) 
+{
+	RECOIL_DETAILED_TRACY_ZONE;
+	LUA_CALL_IN_CHECK(L);
+	luaL_checkstack(L, 8, __func__);
```

Comment:
There's 6 values on the stack (4 args, function, debug traceback)
```suggestion
	luaL_checkstack(L, 6, __func__);
```